uptane/services/inventorydb.py

```diff
@@ -169,4 +84,122 @@ def scrub_filename(fname, expected_containing_dir):
+  inventory_db = {'vehicle_manifests': {},
+                  'ecu_manifests': {},
+                  'public_keys': {
+                                  'vehicle_primaries': {},
+                                  'all_ecus': {}
+                                 }
+                 }
+
+  # Register ECU
+  def register_ecu(self, isPrimary, vin, ecu_serial, public_key):
+    if isPrimary:
+      vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
+      if (vehicle_primary_ids and (vin in vehicle_primary_ids)):
+        # rewrite value
+        (self.inventory_db['public_keys']['vehicle_primaries'])[vin] = \
+           {'ecu_id': ecu_serial, 'public_key': public_key}
+      else:
+        temp_dic = {'ecu_id': ecu_serial, 'public_key': public_key}
+        (self.inventory_db['public_keys']['vehicle_primaries'])[vin] = temp_dic
+    else:
+      ecu_serial_ids = self.inventory_db['public_keys']['all_ecus']
+      if (ecu_serial_ids and (ecu_serial in ecu_serial_ids)):
+        (self.inventory_db['public_keys']['all_ecus'])[ecu_serial] = public_key
+      else:
+        (self.inventory_db['public_keys']['all_ecus'])[ecu_serial] = public_key
+
+    is_registered = 1
+
+
+  # Save vehicle manifest
+  def save_vehicle_manifest(self, vin, signed_vehicle_manifest):
+    """
+    Given a manifest of form
+    uptane.formats.SIGNABLE_VEHICLE_VERSION_MANIFEST_SCHEMA, save it in an index
+    by vin, and save the individual ecu attestations in an index by ecu serial.
+    """
+    vehicle_primary_ids = self.inventory_db['public_keys']['vehicle_primaries']
+    ecu_serial_ids = self.inventory_db['public_keys']['all_ecus']
+    if len(vehicle_primary_ids) == 0 or len(ecu_serial_ids) == 0:
+      raise uptane.UnknownECU("InventoryDataBase has not registered yet!")
+
+    vehicle_ids = self.inventory_db['vehicle_manifests']
+    if (vehicle_ids and (vin in vehicle_ids)):
+      (self.inventory_db['vehicle_manifests'][vin]).append(signed_vehicle_manifest)
+    else:
+      vm_list = []
+      vm_list.append(signed_vehicle_manifest)
+      vehicle_ids[vin] = vm_list
+      self.inventory_db['vehicle_manifests'] = vehicle_ids
+
+
+  # Get vehicle manifest
+  def get_vehicle_manifest(self, vin):
+    # Check arg format
+    vehicle_ids = self.inventory_db['vehicle_manifests']
+    if (vehicle_ids and (vin in vehicle_ids)):
+      return self.inventory_db['vehicle_manifests'][vin]
+    return False
+
+
+  # Save ECU manifest
+  def save_ecu_manifest(self, ecu_serial, signed_ecu_manifest):
+    uptane.formats.VIN_SCHEMA.check_match(vin)
+    uptane.formats.ECU_SERIAL_SCHEMA.check_match(ecu_serial)
+    uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA.check_match(
+        signed_ecu_manifest)
+
+    ecu_serial_ids = self.inventory_db['ecu_manifests']
+    if (ecu_serials_ids and (ecu_serial in ecu_serials_ids)):
+      (self.inventory_db['ecu_manifests'][ecu_serial]).append(signed_vehicle_manifest)
+    else:
```

Comment:
The next four lines should instead be replaced by `ecu_serial_ids[ecu_serial] = [signed_vehicle_manifest]`